Add back JDBC templates to the GF distribution

OndroMih opened the pull request:
Fixes https://github.com/eclipse-ee4j/glassfish/issues/24167 and possibly other issues.

JDBC templates need to be a ZIP artifact to unpack them into the distribution. This reverts making it a JAR artifact in https://github.com/dmatej/glassfish/commit/a53e2d0a0f9b4e96526c706dbd9132bc08b4eeb1.

WARNING: When merging, please SQUASH the commits and use the message from the last one. The second commit only reverts the first one, and they both can be removed from the history.

OndroMih committed:
#24167 - Add back JDBC templates to the GF distribution
https://github.com/eclipse-ee4j/glassfish/issues/24167
- JDBC templates need to be a ZIP artifact to unpack them into the distribution
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

avpinchuk commented:
Maybe we can adjust assembly descriptor without fallback to `zip`?

OndroMih commented:
> Maybe we can adjust assembly descriptor without fallback to zip?

I tried but it's not that easy. The [glassfish-build-maven-plugin](https://github.com/eclipse-ee4j/glassfish-build-maven-plugin/https://github.com/eclipse-ee4j/glassfish-build-maven-plugin/commit/145a8b9e917312b6453e7cea2717b3eda5715370/master/src/main/java/org/glassfish/build/FeatureSetsDependenciesMojo.java) can either unpack all JARs (with some excluded), or none. 

So we would have to exclude all those that we don't want to unpack or exclude them in the assembly descriptor.

Alternatively, we could unpack the template.jar copied by the glassfish build plugin, probably using the antrun plugin.

But is it worth it? The templates artifact is never used as a JAR, it's always unpacked as a ZIP file, in both featuresets/web and jdbc-ra artifacts.

avpinchuk commented:
@OndroMih, adding this to `full` and `web` assembly descriptor seems to solve the problem

```xml
 <dependencySets>
        <dependencySet>
            <includes>
                <include>org.glassfish.main.jdbc:templates:jar:${artifact.version}</include>
            </includes>
            <unpack>true</unpack>
            <unpackOptions>
                <excludes>
                    <exclude>META-INF/**</exclude>
                </excludes>
            </unpackOptions>
            <outputDirectory>${install.dir.name}</outputDirectory>
        </dependencySet>
    </dependencySets>
```

That's correct way?

OndroMih committed:
#24167 - Add back JDBC templates to the GF distribution
https://github.com/eclipse-ee4j/glassfish/issues/24167
- JDBC templates need to be unpacked by the assembly plugin
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Revert "#24167 - Add back JDBC templates to the GF distribution https://github.com/eclipse-ee4j/glassfish/issues/24167 - JDBC templates need to be a ZIP artifact to unpack them into the distribution Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>"

This reverts commit 774a3565f7b5f44aae754a7b55649926ff3f14fd.

OndroMih commented:
Yes, this works, @avpinchuk, thanks! I didn't know that the Maven Assembly plugin can also unpack ZIP files. I reverted my previous commit and applied your suggestion.

dmatej commented:
Aha, I was scared by an e-mail writing about a revert, which would break embedded again. Now it seems alright.
